Sources/TuistGenerator/Linter/StaticProductsGraphLinter.swift

```diff
@@ -52,72 +55,101 @@ class StaticProductsGraphLinter: StaticProductsGraphLinting {
-            linkingNode.target.canLinkStaticProducts()
+        guard case let ValueGraphDependency.target(targetName, targetPath) = dependency,
+            let dependencyTarget = graphTraverser.target(path: targetPath, name: targetName),
+            dependencyTarget.target.canLinkStaticProducts()
         else {
             return results
         }
 
         while let staticProduct = results.unlinked.popFirst() {
-            results.linked[staticProduct, default: Set()].insert(linkingNode)
+            results.linked[staticProduct, default: Set()].insert(dependency)
         }
 
         cache.cache(results: results,
-                    for: node)
+                    for: dependency)
 
         return results
     }
 
-    private func staticDependencyWarning(staticProduct: GraphNode,
-                                         linkedBy: Set<TargetNode>) -> [StaticDependencyWarning]
+    private func staticDependencyWarning(staticProduct: ValueGraphDependency,
+                                         linkedBy: Set<ValueGraphDependency>,
+                                         graphTraverser: GraphTraversing) -> [StaticDependencyWarning]
     {
         // Common dependencies between test bundles and their host apps are automatically omitted
         // during generation - as such those shouldn't be flagged
         //
         // reference: https://github.com/tuist/tuist/pull/664
-        let apps: Set<GraphNode> = linkedBy.filter { $0.target.product == .app }
-        let hostedTestBundles = linkedBy
-            .filter { $0.target.product.testsBundle }
-            .filter { $0.dependencies.contains(where: { apps.contains($0) }) }
+        let apps: Set<ValueGraphDependency> = linkedBy.filter { (dependency) -> Bool in
+            guard case let ValueGraphDependency.target(targetName, targetPath) = dependency else { return false }
+            guard let target = graphTraverser.target(path: targetPath, name: targetName) else { return false }
+            return target.target.product == .app
+        }
+        let hostedTestBundles = linkedBy.filter { (dependency) -> Bool in
+            guard case let ValueGraphDependency.target(targetName, targetPath) = dependency else { return false }
+            guard let target = graphTraverser.target(path: targetPath, name: targetName) else { return false }
+
+            let isTestsBundle = target.target.product.testsBundle
+            let hasHostApp = graphTraverser.dependencies[dependency, default: Set()].contains(where: { apps.contains($0) })
+            return isTestsBundle && hasHostApp
+        }
 
         let links = linkedBy.subtracting(hostedTestBundles)
 
         guard links.count > 1 else {
             return []
         }
 
-        let sortedLinks = links.sorted(by: { $0.name < $1.name })
         return [
             .init(staticProduct: staticProduct,
-                  linkingNodes: sortedLinks),
+                  linkingDependencies: Array(links)),
```

Comment:
looks like the sorting has been omitted - I believe the intention was to ensure the results were stable when producing those lint errors